### PR TITLE
Add school model with api

### LINKED ISSUE #24: Feature: Create Model "School" with CRUD
## Changes
- School
  - `name` : string
  - `timezone`: string ([List of tz database time zones](https://en.wikipedia.org/wiki/List_of_tz_database_time_zones))
- CRUD
  - Create [ONLY ADMINS]
  - Read [All Users]
  - Update [ONLY ADMINS]
  - Delete [ONLY ADMINS]
- Update Postman Documentation 
- Update the Access Control Excel Sheet

### PR DESCRIPTION
**Pull request by jasonqiu212**
## Changes
- Add school model.
- Add school controller, service, repository.
- Add school routes.
- Add school migration and seed data.
- Updated user access flow in `README.md`.
- Add associations for `User`, `Resource`, and `Subscription` to `School`.

## Testing
- Updated Postman collection with `School`.
- Use Postman and test all routes in the `School` Collection.

## Issues
- Resolves #24

### CODE REVIEW

File: src/db/seeders/20220514012040-sample-schools.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+'use strict';
+
+module.exports = {
+  up: async queryInterface => {
+    return await queryInterface.bulkInsert('schools', [
+      {
+        name: 'Dulwich College Shanghai Pudong',
```
Comment (ShenyiCui) [inline src/db/seeders/20220514012040-sample-schools.js:7]: Change these names to just their regions: 
Shanghai Pudong, 
Singapore

File: src/models/School.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+      {
+        tableName: School.getTableName()!,
+        sequelize,
+      }
+    );
+  }
+
+  public static associate(models: Models) {
+    School.hasMany(models.User, {
+      onDelete: 'CASCADE',
```
Comment (ShenyiCui) [inline src/models/School.ts:61]: Oh wow, wait so do these handle transactions for us? I've never seen this before

**Review by ShenyiCui — APPROVED** [review]
lgtm! just a few small changes to make. Check that CASCADE helps with perform a transaction implicitly, if so then its good to go!

File: src/db/seeders/20220514012040-sample-schools.js (hunk 1 above)
Comment (ShenyiCui) [reply src/db/seeders/20220514012040-sample-schools.js:7]: Also please add a Beijing hahaha Shanghai was our rival school

**Review by ShenyiCui — CHANGES_REQUESTED** [review]
As mentioned on telegram please edit your user access flow
Screenshot 2022-06-17 at 1 08 00 AM

If students are the room admin then they should have the ability to delete and update the resource. So it's basically what we discussed before, there will be a separate middleware to check if they own the resource or if they're an admin

File: src/db/seeders/20220514012040-sample-schools.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+        timezone: 'Asia/Shanghai',
+        createdAt: new Date(),
+        updatedAt: new Date(),
+      },
+      {
+        name: 'Dulwich College Singapore',
+        timezone: 'Asia/Singapore',
```
Comment (ShenyiCui) [inline src/db/seeders/20220514012040-sample-schools.js:14]: Can you create a new type or interface called Timezone and also a Constant to store all the available timezones?

File: src/db/seeders/20220514012040-sample-schools.js (hunk 1 above)
Comment (jasonqiu212) [reply src/db/seeders/20220514012040-sample-schools.js:7]: HAHA same, they beat us in vball .-.

File: src/db/seeders/20220514012040-sample-schools.js (hunk 3 above)
Comment (jasonqiu212) [reply src/db/seeders/20220514012040-sample-schools.js:14]: okok good thinking

**Review by ShenyiCui — APPROVED** [review]
lgtm

**Pull request merged**